Stacked @card PR : Base PR

valayDave opened the pull request.

valayDave committed:
Base card PR With minimal Diffs

valayDave committed:
Fixed bug in the `_NamespacePath`

valayDave committed:
Fixing bug fix.

valayDave committed:
Fixing card datastore root related changes.

tuulos commented on metaflow/plugins/cards/card_decorator.py:
```diff
@@ -0,0 +1,192 @@
+            cmd += ["--timeout", str(self.attributes["timeout"])]
+
+        self._run_command(cmd, os.environ)
+        # do nothing on failure as we create an error card
```
if the subprocess has a hard crash, nothing is shown currently - not even an error card. You could handle soft crashes quietly with the error card, which can return 0, but if the subprocess fails hard, you could print an error on stderr with the error code

tuulos commented on metaflow/plugins/cards/card_decorator.py:
```diff
@@ -0,0 +1,192 @@
+            rep = subprocess.check_output(
+                cmd,
+                env=env,
+                stderr=subprocess.STDOUT,
```
you could set `timeout=` here too. Currently you rely on the subprocess to behave nicely but that's not guaranteed - especially with custom cards

valayDave committed:
Fixing `metaflow_card_cache` creation

valayDave committed:
Fixed namespace mismatch + refactor
- `follow_resumed` arg refactor

valayDave committed:
changes ported from `metaflow-cards-main-pr`
- simplifying function args
- removed `metaflow_cards` ns package
- integrated card s3 root changes to batch + eks
- removed py2 tests

valayDave committed:
importing cli changes requested by romain

valayDave committed:
removed `CARDS` from `_expected_extensions`
